batch/batch/server/server.py

```diff
@@ -791,35 +752,48 @@ def from_record(record):
                                        ttl=ttl,
                                        is_open=is_open,
                                        userdata=json.dumps(userdata),
-                                       user=user)
+                                       user=user,
+                                       deleted=False)
 
         batch = Batch(id=id, attributes=attributes, callback=callback,
-                      ttl=ttl, is_open=is_open, userdata=userdata, user=user)
+                      ttl=ttl, is_open=is_open, userdata=userdata, user=user,
+                      state='running', complete=False, deleted=False)
         batch.close_ttl()
         return batch
 
-    def __init__(self, id, attributes, callback, ttl, is_open, userdata, user):
+    def __init__(self, id, attributes, callback, ttl, is_open, userdata, user,
+                 state, complete, deleted):
         self.id = id
         self.attributes = attributes
         self.callback = callback
         self.ttl = ttl
         self.is_open = is_open
         self.userdata = userdata
         self.user = user
+        self.state = state
+        self.complete = complete
+        self.deleted = deleted
 
     async def get_jobs(self):
         return [Job.from_record(record) for record in await db.jobs.get_records_by_batch(self.id)]
 
     async def cancel(self):
+        await self.close()
         jobs = await self.get_jobs()
         for j in jobs:
             await j.cancel()
 
     async def delete(self):
+        # Batch is deleted from db in polling loop once all jobs are complete
+        self.is_open = False
+        self.deleted = True
         jobs = await self.get_jobs()
```

Comment:
Getting the jobs between setting the local fields and the database fields worries me.

I also think we should set the local fields AFTER the database fields.  If the database write fails, they will be out of sync.

Reply:
My concern was if the purge of records happens after the batch record in the database is set to deleted but before we get the job records out.

Reply:
Oh, good point.

What about just calling self.cancel() and then set deleted=True in the db.